### PR TITLE
Fix stopping and starting application

### PR DESCRIPTION
**Pull request by lauracavaliere**
With this PR we want to fix the error generated by the deployment if we stop an application and start it again.
The error was related to the fact that the `appsAway_stopApp.sh` removed the registry and thus when calling the `appsAway_startApp.sh` the registry didn't exist anymore (it was created by `appsAway_setupCluster.sh`).

To fix this, we:
- create the registry with `docker-compose`; this implied changing the structure of `appsAway_registryLaunch.yml` which has to be the one required by `docker-compose`
- remove the registry in a new script `appsAway_cleanupCluster.sh` which is called when we close the GUI (i.e. at the end of the script generated by robot-bazaar)

This implementation requires changing the website to update the generation of the `demoName_setup.sh`.

### CODE REVIEW

**Review by vvasco — APPROVED** [review]
LGTM

**Comment by lauracavaliere** [comment]
While testing these changes, we noticed that the way we were checking if some containers were already running on port 5000 was wrong. 

We fixed this in the [last commit](https://github.com/icub-tech-iit/appsAway/pull/244/commits/ba88d34711d1860dc060ec4159e4519aef90843f): we retrieve the list of container ids, inspect them and grep port 5000. If the output of this is empty, it means no registry is running on the port, otherwise, the port is busy and we exit from the script.

**Comment by valegagge** [comment]
With @ilaria-carlini and @lauracavaliere created the script file `appsAway_setupRegistry.sh` which takes care to check if it is necessary to create the local registry and eventually it sets registry IP address and port in the docker daemon file and then launches the local registry by docker compose.

Tested on a single node and it works fine. 

Please see [here](https://github.com/icub-tech-iit/appsAway/pull/244/commits/b9327c3ef203b85abdbf2936113cc2841b8c6c6b) the improvement.

File: scripts/appsAway_cleanupCluster.sh [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    if [[ ${REGISTRY_UP_FLAG} == false ]]
+    then
+        log "Removing registry service"
+        ${_DOCKER_COMPOSE_BIN_CONSOLE} -f appsAway_registryLaunch.yml down
+    else 
+        log "No registry has been created by this deployment. No need to remove it."
+    fi
+}
+
+main()
```
Comment (AlexAntn) [inline scripts/appsAway_cleanupCluster.sh:132]: I have some questions regarding this PR:

1) Is this script meant to clean up the entire cluster? if yes, we need to make sure it also cleans up all the running containers (like running the 'appsAway_stopApp.sh' script). 
2) If this script is meant to clean up only the registry, I think the name is a bit misleading. Furthermore, in the PR in code, we should include also the 'appsAway_stopApp.sh' script in the demo setup script downloaded from the website.

Other than this it LGTM

**Comment by vvasco** [comment]
Let's also remember to update the related PR in code to call the `appsAway_setupRegistry.sh` from the cluster setup created from the website.

**Review by AlexAntn — COMMENTED** [review]
I have some questions regarding the `appsAway_cleanupCluster.sh` script:

1) Is this script meant to clean up the entire cluster? if yes, we need to make sure it also cleans up all the running containers (like running the `appsAway_stopApp.sh` script). 
2) If this script is meant to clean up only the registry, I think the name is a bit misleading. Furthermore, in the PR in code, we should include also the `appsAway_stopApp.sh` script in the demo setup script downloaded from the website.

Other than this it LGTM

**Comment by valegagge** [comment]
> I have some questions regarding the `appsAway_cleanupCluster.sh` script:
> 
> 1. Is this script meant to clean up the entire cluster? if yes, we need to make sure it also cleans up all the running containers (like running the `appsAway_stopApp.sh` script).
> 2. If this script is meant to clean up only the registry, I think the name is a bit misleading. Furthermore, in the PR in code, we should include also the `appsAway_stopApp.sh` script in the demo setup script downloaded from the website.
> 
> Other than this it LGTM

I agree with you, now the name of the file is a bit misleading, but we think that in this file we could add some other cleanup, like stop and remove containers, exit from the swarm, etc.
We already planning to clean the entire cluster when we finished the deployment, but, currently, we have no time to do.
So in the end, if we prefer to have a single script to clean registry we could rename the file, else we can leave as it is.

**Comment by vvasco** [comment]
> > I have some questions regarding the `appsAway_cleanupCluster.sh` script:
> > 
> > 1. Is this script meant to clean up the entire cluster? if yes, we need to make sure it also cleans up all the running containers (like running the `appsAway_stopApp.sh` script).
> > 2. If this script is meant to clean up only the registry, I think the name is a bit misleading. Furthermore, in the PR in code, we should include also the `appsAway_stopApp.sh` script in the demo setup script downloaded from the website.
> > 
> > Other than this it LGTM
> 
> I agree with you, now the name of the file is a bit misleading, but we think that in this file we could add some other cleanup, like stop and remove containers, exit from the swarm, etc. We already planning to clean the entire cluster when we finished the deployment, but, currently, we have no time to do. So in the end, if we prefer to have a single script to clean registry we could rename the file, else we can leave as it is.

We can leave it as it is for now and extend it when we are going to clean up when the deployment exits.

**Pull request merged**